src/Pinion.ts

```diff
@@ -108,10 +108,14 @@ class Pinion {
         } catch (caughtError) {
           console.error(caughtError);
         }
+        break;
       }
       case PIN_HASH: {
         if (!ipfsHash) {
-          log('PIN_HASH: no ipfsHash given: %O', message.data);
+          if (log.enabled) {
+            // Only stringify when absolutely necessary
```

Comment:
Can we add this to the `REPLICATE` case below and the 'could not parse pinner message' error above?

Reply:
Oh, yeah sure, I missed those!

Reply:
Done